internal/bzlmod/go_deps.bzl

```diff
@@ -300,6 +313,44 @@ build_naming_convention) or "gazelle:proto <value>" (for build_proto_file_mode)
 to its 'directives' attribute.
 """ + format_module_file_fixup(root_fixups))
 
+def with_replaced_or_new_fields(_struct, **replacements):
+    """Provides a shallow copy of a structure with replacements and/or new fields
+
+    Args:
+        _struct: structure to shallow copy.
+        **replacements: kwargs for fields to either replace or add to the new struct.
+
+    Returns:
+        The resulting updated structure.
+    """
+    new_struct_assignments = {}
+
+    # structs have certain built-in methods or attributes that should not
+    # be overwritten. Load these by calling `dir` on an empty struct.
+    # e.g. ["to_proto", "to_json"]
+    protected_attributes = dir(struct())
+    for key in dir(_struct):
```

Comment:
This for loop could just be a dict comprehension.

Reply:
Yes. I also decided to add a Unit Test for this function since it's functionality is important.

Reply:
While you are at it, could you move it to the new utils.bzl file?